api/types/types.go

```diff
@@ -657,15 +657,16 @@ type Runtime struct {
 	Path string   `json:"path"`
 	Args []string `json:"runtimeArgs,omitempty"`
 
-	// This is exposed here only for internal use
-	// It is not currently supported to specify custom shim configs
-	Shim *ShimConfig `json:"-"`
+	Shim *ShimConfig `json:"shim"`
```

Comment:
This type is also used in the `Info` type, which is used for the `/info` response; https://github.com/moby/moby/blob/9ccedde9703bbef70881a0e8a3006310b7d3d788/api/types/types.go#L302

So;

- some changes are needed in the [`Runtime` definition in the Swagger file](https://github.com/moby/moby/blob/9ccedde9703bbef70881a0e8a3006310b7d3d788/api/swagger.yaml#L5461-L5486)
- and an entry to be added to the [API version-history.md](https://github.com/moby/moby/blob/9ccedde9703bbef70881a0e8a3006310b7d3d788/docs/api/version-history.md)

Reply:
Fixed, please take a look again.